pages/index.vue

```diff
@@ -109,63 +114,87 @@ import addTask from '~/components/addTask.vue'
     async getTasks() {
-      /***
-       * @todo Fetch the tasks created by the user and display them.
-       * @todo also the function to display a single new task added
-       * @hints use store and set loading true
-       * @caution you have to assign new value to todos for it to update
-       */
+    this.$axios({
+      headers: {
+        Authorization: 'Token ' + this.$store.getters.token,
+      },
+      url: 'todo/',
+      method: 'get',
+    })
+      .then(({ data }) => {
+        this.loading = false
+        this.todos = data.map((todo) => {
+          return {
+            ...todo,
+            editing: false,
+          }
+        })
+      })
+      .catch(() => {
+        this.$toast.error('Error in fetching the tasks')
+      })
     },
-    /**
-     * Function to update a single todo
-     * @argument {number} _index - index of element to update in todos array
-     * @argument {number} _id - id of todo obtained from API
-     * @todo Complete this function.
-     * @todo 1. Send the request to update the task to the backend server.
-     * @todo 2. Update the task in the dom.
-     */
-    updateTask(_index, _id) {},
-    /**
-     * toggle visibility of input and buttons for a single todo
-     * @argument {number} index - index of element to toggle
-     * @todo add in bindings in dom so that 'hideme' class is dynamic or use conditional rendering
-     * @hint read about class bindings in vue
-     */
+
+    updateTask(_index, _id) {
+    if (this.todos[_index].title === '') {
+       this.$toast.error('Task cannot be empty!!')
+     } else {
+       this.$axios({
+         headers: {
+           Authorization: 'Token ' + this.$store.getters.token,
+         },
+         url: 'todo/' + _id + '/',
+         method: 'patch',
+         data: {
+           title: this.todos[_index].title,
+         },
+       })
+         .then(() => {
+           this.todos[_index].editing = false
+           this.$toast.success('Task updated successfully')
+         })
+         .catch(() => {
+           this.$toast.error('Error in updating the tasks')
+         })
+     }
+    },
+
     editTask(index) {
       this.todos[index].editing = !this.todos[index].editing
     },
-    /**
-     * Function to delete a single todo
-     * @argument {number} _index - index of element to update in todos array
-     * @argument {number} _id - id of todo obtained from API
-     * @todo Complete this function.
-     * @todo 1. Send the request to delete the task to the backend server.
-     * @todo 2. Remove the task from the dom.
-     */
-    deleteTask(_index, _id) {},
+
+    deleteTask(_index, _id) {
+    this.$axios({
+        headers: {
+          Authorization: 'Token ' + this.$store.getters.token,
+        },
+        url: 'todo/' + _id + '/',
+        method: 'delete',
+      })
+        .then(() => {
+          this.$toast.success('Task deleted successfully')
+          this.todos = this.todos.filter((todo) => todo.id !== _id)
```

Comment:
splice would have been a better choice

Reply:
Splice skipped my mind at that moment! Will remember it hereafter.